src/thor/trace_route_action.cc

```diff
@@ -46,25 +46,27 @@ namespace thor {
 /*
  * The trace_route action takes a GPS trace and turns it into a route result.
  */
-odin::TripLeg thor_worker_t::trace_route(valhalla_request_t& request) {
-
+std::list<odin::TripLeg> thor_worker_t::trace_route(valhalla_request_t& request) {
   // Parse request
   parse_locations(request);
   parse_costing(request);
   parse_measurements(request);
 
+  std::list<odin::TripLeg> trip_paths;
+
   // Initialize the controller
-  odin::TripLeg trip_path;
   AttributesController controller;
 
   switch (request.options.shape_match()) {
     // If the exact points from a prior route that was run against the Valhalla road network,
     // then we can traverse the exact shape to form a path by using edge-walking algorithm
     case odin::ShapeMatch::edge_walk:
       try {
-        trip_path = route_match(request, controller);
-        if (trip_path.node().size() == 0) {
-          throw std::exception{};
+        trip_paths = route_match(request, controller);
```

Comment:
i dont think this should ever return more than one. so this can probably be simplified to:

    if(trip_paths.empty() || trip_paths.front().node().empty())